### DISCUSSION TITLE
Reverb & Delay effects

### DISCUSSION THREAD

**Discussion by amesser** [DatanoiseTV/PicoDSP-Hardware#3] [Q&A]
Hi,

I just started to play a bit with RP2040 for effect stuff and was pointed to your project. I have reviewed your code and wondering how the reverb/delay stuff comes in. As far as I understand the "vult" code for reverb is not used at the moment? But the SoundCloud sounds seem to have some reverb?

Did you had a look at the performance? My plan was to use super simple RP2040 circuit using onchip ADC and a Sigma-Delta-DAC for some echo/reverb/chorus stuff. Turned out that its not possible to have the pio units do the PCM to PDM (Sigma-Delta), must be done on Cores.  In the end, this probably means, it may not have enough horsepower to do this like I have planned it.

**Comment by DatanoiseTV** [comment]
Hi,
I've had some delay and simple reverb used in the Vult code, but it is no longer used to save resources. We only have 264KB of RAM, so delays wouldn't be very long at 44100 or 22050Hz sample rate.

**Comment by amesser** [comment]
Hi, thanks for confirm that I have understood the source correctly. So I have to go back to my idea. I already withdrawn the idea of a full featured fx module built on RP2040 since I have come across some limitations of RP2040 which would waste too much of its horsepower for non-fx stuff. My current plan is to implement 4 delay lines similar to what is found in PT2399 but with two taps. A prototype with one is already working. They use Sigma-Delta bitstreams. This helps to save memory while still maintaining somewhat of ADC resolution. I still need to improve this a bit.
